docs/build/spells.md

```diff
@@ -0,0 +1,126 @@
+
+This service is intended to be used as a template and instantiated by a node on each new spell creation.
+
+## Managing spells
+Nox provides user API for interacting with spells.
+
+Users are able to control their spells' lifecycle: install new ones, update them, and remove them. The developer experience is provided by the means of Fluence CLI. 
+
+Installing spells include providing the following: 
+- The spell script, which will be executed
+- The list of trigger events (aka the trigger configuration) on which to run the spell
+- The initial state of the spell (e.g. initial state of spell's KV)
+
+After installation, the developer is able to update the initial settings: the script, the trigger configuration, and the state KV storage.
+
+Also, the developer is able to remove their spells, cleaning the state from the peer.
+
+Users can check their spells' state: 
+- inspect their internal state (the KV storage, the script, the trigger config)
+- check if they’re currently subscribed to some triggers
+- obtain their peer identity: spell_id and worker_id.
+
+## Events that can trigger spells
+Nox triggers spells according to their configuration. Currently Nox provides the following event triggers:
+- Timer/CRON Triggers: run spells at a specified time or within a specified period.
+Users should be able to specify the following:
+    - when to run the spell,
+    - when to unsubscribe the spell from timer events,
+    - with what period to run the spell.
+- Connection Triggers: run spells on connection/disconnection of some peers to the peer.
+
+All other trigger types can be implemented based on the timer triggers & the spell KV storage (blockchain event trigger, database event trigger, HTTP event trigger, etc)
+
+## Creating and updating spells
+
+Spells can be created and managed with Fluence CLI. 
+
+Let's create a new spell in a recently scaffolded Fluence CLI project by running `fluence spell new`:
+```bash
+quickstart $ fluence spell new
+? Enter spell name 5min_spell
+Successfully generated template for new spell at /Users/arete/Documents/fluence/quickstart/src/spells/1min_spell
+? Do you want to add spell 5min_spell to a default worker defaultWorker Yes
+Added 1min_spell to defaultWorker
+```
+
+Lets check what have been generated:
+- a new spell directory was created under src/spells -- 5min_spell:
+```bash
+.
+└── 1min_spell
+    ├── spell.aqua
+    └── spell.yaml
+```
+
+It contains aqua script that will be executed and the spell.yaml file that specifies the trigger configuration for the spell.
+
+If we take a look at the spell.aqua file we see the `spell()` function that is executed when the spell is triggered:
+```aqua
+import Op, Debug from "@fluencelabs/aqua-lib/builtin.aqua"
+import Spell from "@fluencelabs/spell/spell_service.aqua"
+
+func spell():
+    msg = "Spell is working!"
+    str <- Debug.stringify(msg)
+    Spell "worker-spell"
+    Spell.list_push_string("logs", str)
```

Comment:
You should use [this](https://github.com/fluencelabs/spell/blob/main/src/aqua/spell/spell_service.aqua#L139). And also explain why you store logs in installation-spell (with some link to deep-dive explanation later)

Reply:
I suggest to rewrite it like this:
```aqua
import Op, Debug from "@fluencelabs/aqua-lib/builtin.aqua"
import Spell from "@fluencelabs/spell/spell_service.aqua"
func spell():
    msg = "Spell is working!"
    str <- Debug.stringify(msg)
    Spell "self"
    -- also it can be resolved by Spell "spell"
    Spell.set_string("my_key", str)
    -- then we will log the same message to installation-spell
    Spell "worker-spell"
    Spell.store_log(str)
```
To check in `main.aqua`, we will define a function `get_my_keys`:
```aqua
aqua Main

import Console, Debug from "@fluencelabs/aqua-lib/builtin.aqua"
import "@fluencelabs/aqua-lib/subnet.aqua"
import Spell from "@fluencelabs/spell/spell_service.aqua"

use "deals.aqua"

export get_my_keys

func get_my_keys() -> []string:
    deals <- Deals.get()
    dealId = deals.defaultWorker!.dealIdOriginal
    results: *string
    on HOST_PEER_ID:
        subnet <- Subnet.resolve(dealId)
    if !subnet.success:
        Console.print(["Failed to resolve subnet: ", subnet.error])

    for w <- subnet.workers:
        if w.worker_id != nil:
            on w.worker_id! via w.host_id:
                Spell "5min_spell"
                res <- Spell.get_string("my_key")
                results <- Debug.stringify([res.str, "on worker ", w.worker_id!, " on host ", w.host_id])

    <- results
```

Output:
```
fluence run -f "runDeployedServices()"
Connecting to kras relay: /dns4/8-kras.fluence.dev/tcp/9000/wss/p2p/12D3KooWEFFCZnar1cUJQ3rMWjvPQg6yMV2aXWs2DkJNSRbduBWn
Connected
[
  "[\"\\\"Spell is working!\\\"\",\"on worker \",\"12D3KooWMaEjJ7ren3JvVgAtrzm6aPPtSZMPTcQu8wD9wZ5krQe5\",\" on host \",\"12D3KooWSD5PToNiLQwKDXsu8JSysCwUt8BVUJEqCHcDe7P5h45e\"]",
  "[\"\\\"Spell is working!\\\"\",\"on worker \",\"12D3KooWPkUpq6U2FpMbviNystGD7qmPa9yM8oR2dv1WkAd4jtkW\",\" on host \",\"12D3KooWR4cv1a8tv7pps4HH6wePNaK6gf1Hww5wcCMzeWxyNw51\"]",
  "[\"\\\"Spell is working!\\\"\",\"on worker \",\"12D3KooWSY8ZFEoAXmHYkYheSRcDB7FwmtmNz8iHYmZjcArDYhmc\",\" on host \",\"12D3KooWHLxVhUQyAuZe6AHMB29P7wkvTNMn7eDMcsqimJYLKREf\"]"
]
```

And then run `deal logs`

Reply:
Why is it marked as resolved, but still uses `list_push_string`?

Reply:
[DXJ-514 Use Spell.store_log in fluence spell new template](https://linear.app/fluence/issue/DXJ-514/use-spellstore-log-in-fluence-spell-new-template) added an issue to resolve that in fcli

Reply:
@meph what's the update status on this? as it stands, code will throw an error.